Backend/src/main/java/ca/mcgill/cooperator/controller/StudentController.java

```diff
@@ -204,6 +205,26 @@ public CoopDto getCurrentStudentCoop(@PathVariable int id) {
         return null;
     }
 
+    /**
+     * Gets the upcoming Coop(s) for the specified Student
+     *
+     * @param id
+     * @return Coop(s) or empty list if no upcoming Coops
+     */
+    @GetMapping("/{id}/upcoming-coops")
+    public List<CoopDto> getUpcomingStudentCoops(@PathVariable int id) {
+        Student s = studentService.getStudentById(id);
+        Set<Coop> coops = s.getCoops();
+        List<CoopDto> result = new ArrayList<>();
+        for (Coop c : coops) {
+            if (c.getStatus() == CoopStatus.FUTURE
+                    || c.getStatus() == CoopStatus.UNDER_REVIEW
+                    || c.getStatus() == CoopStatus.INCOMPLETE)
```

Comment:
how is incomplete an upcoming coop

Reply:
I was thinking `INCOMPLETE` meant a student added a new co-op but it got rejected by an Admin; technically it's still upcoming

Reply:
Add Rejected and Failed as enums instead of INCOMPLETE